File: flutter_app/lib/fable_dart/Library.dart
// ignore_for_file: camel_case_types, constant_identifier_names, non_constant_identifier_names, unnecessary_this
import './GSTFlow.Core/Library.dart' as library_2;
import './GSTFlow.Emit/Library.dart' as library_1;
import './GSTFlow.Rules/Library.dart' as library;

library.CompilationResult compileInvoice(library.RawInvoice invoice, String hash) => library.Compiler_compile(invoice, hash);

String emitSummary(library_2.GSTCanonicalIR ir) => library_1.emitSummaryJson(ir);

String emitReport(library_2.GSTCanonicalIR ir) => library_1.emitValidationReport(ir);

File: GSTFlow.Dart/Library.fs
module GSTFlow.Dart.API

open GSTFlow.Core
open GSTFlow.Core.Verification
open GSTFlow.Rules
open GSTFlow.Emit

// We expose the core RawInvoice and Compiler to Dart.
// Fable will generate Dart classes for RawParty, RawInvoiceItem, RawInvoice, and VerdictEnvelope.
// The Flutter app can construct these objects directly using Dart syntax, and then pass them here.

let compileInvoice (invoice: RawInvoice) (hash: string) =
    Compiler.compile invoice hash

let emitSummary (ir: GSTCanonicalIR) =
    Generators.emitSummaryJson ir

let emitReport (ir: GSTCanonicalIR) =
    Generators.emitValidationReport ir
